src/javascript/ContentEditor/adaptEditFormData.js

```diff
@@ -145,12 +145,16 @@ const getUsagesInfo = nodeData => {
     ), {}));
 };
 
-export const adaptEditFormData = (data, lang, t) => {
+export const adaptEditFormData = (data, lang, t, contentEditorConfigContext) => {
     const nodeData = data.jcr.result;
     const sections = adaptSections(data.forms.editForm.sections);
 
+    // Take expanded section from existing config, or use the one from the form definition.
+    const expandedSections = contentEditorConfigContext.expandedSections ? contentEditorConfigContext.expandedSections : (sections ? sections.reduce((result, section) => ({...result, [section.name]: section.expanded}), {}) : {});
```

Comment:
It's a bit complex to read, can you move `sections ? sections.reduce((result, section) => ({...result, [section.name]: section.expanded}), {}) : {}` into a function and use it here and also in the adaptCreateFormData please?

Reply:
indeed, thank you for the review